### PR TITLE
Flow docs: document proxy requirements for otelcol.exporter.otlp

### PR DESCRIPTION
**Pull request by rfratto**

### CODE REVIEW

**Review by karengermond — APPROVED** [review]
LGTM! One grammatical issue that perhaps you could fix in another PR...

File: docs/sources/flow/reference/components/otelcol.exporter.otlp.md [hunk 1]
```diff
@@ -89,7 +89,27 @@ The `balancer_name` argument controls what client-side load balancing mechanism
...
+
+* `HTTPS_PROXY`
+* `NO_PROXY`
+
+The `HTTPS_PROXY` environment variable specifies a URL to use for proxying
+requests. Connections to the proxy are established via [the `HTTP CONNECT`
+method][HTTP CONNECT].
+
+The `NO_PROXY` environment variable is an optional list of comma-separated
+hostnames for which the HTTPS proxy should _not_ be used. Hostnames can be
```
[the remark this answers is not in the document]
Comment (rfratto) [reply docs/sources/flow/reference/components/otelcol.exporter.otlp.md:102]: Does the singular form make sense in the first case? That would make it 

> The `NO_PROXY` environment variable is an optional list of comma-separated a hostname ... 

The idea is that `NO_PROXY` is set to values like `example.com,other-example.com,final-example.com`

File: docs/sources/flow/reference/components/otelcol.exporter.otlp.md [hunk 2]
```diff
@@ -89,7 +89,27 @@ The `balancer_name` argument controls what client-side load balancing mechanism
...
+provided as an IP address (`1.2.3.4`), an IP address in CIDR notation
+(`1.2.3.4/8`), a domain name (`example.com`), or `*`. A domain name matches
+that name and all subdomains. A domain name with a leading "." (`.example.com`)
```
[the remark this answers is not in the document]
Comment (rfratto) [reply docs/sources/flow/reference/components/otelcol.exporter.otlp.md:105]: What about saying "that domain and all subdomains" here?

**Review by tpaschalis — APPROVED** [review]
LGTM! For people who aren't using Flow today, do you think we should sneak this explanation to the bottom of traces_cofig, too?

It won't read _great_, but it would be a reference point. Otherwise, we can keep it in mind as an item for a possible FAQ page.

**Comment by rfratto** [comment]
@tpaschalis Sure, I'm making the other suggested changes quickly before merging

**Comment by rfratto** [comment]
> For people who aren't using Flow today, do you think we should sneak this explanation to the bottom of traces_cofig, too?

I played around with this, but I'd like to leave it out for now since traces_config will need a slightly more generic message in case you are using the HTTP exporter and not the gRPC one.

**Pull request merged**